src/test/ui/in-band-lifetimes/E0687.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![allow(warnings)]
+#![feature(in_band_lifetimes)]
+
+fn foo(x: fn(&'a u32)) {}
+
+fn bar(x: &Fn(&'a u32)) {}
```

Comment:
OK, one last set of tests I would like to see -- I didn't notice this particular scenario. I'm 99.9% sure it works correctly, but it seems like we might as well be complete!

```rust
fn bar<F>(x: &F)
    where F: Fn(&'a u32) { } //~ ERROR

fn bar(x: &impl Fn(&'a u32)) //~ ERROR
```

But:

```rust
fn bar<F>(x: &F) // OK
    where F: SomeTrait<'a>

fn bar(x: &impl SomeTrait<'a>) // OK
```

If you are too busy, I can try adding them, since I'd love to r+ ASAP. =)

Reply:
Done.